src/Controller/TagCloudController.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace App\Controller;
+
+use App\Entity\Searches;
+use App\Repository\SearchesRepository;
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\Persistence\ManagerRegistry;
+use http\Exception\InvalidArgumentException;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+class TagCloudController extends AbstractController
+{
+
+    public function getTopTags(int $qty = 50, SearchesRepository $searches){
+        if ($qty<0) throw new InvalidArgumentException('Bad quantity argument while calling getTopTags function. Argument should be higher than 0.');
+
+        $query = $searches->createQueryBuilder()
```

Comment:
In my opinion, a database query should be in a separate function in SearchesRepository.